src/providers/codeactions.ts

```diff
@@ -172,16 +132,68 @@ export class CodeActions {
                     break
                 }
                 fixString = regexResult[1]
-                replaceRangeWithString(document, range, fixString)
+                this.replaceRangeWithString(document, range, fixString)
                 break
             case 45:
-                replaceMathDelimitersInRange(document, range, '$$', '\\[', '\\]')
+                this.replaceMathDelimitersInRange(document, range, '$$', '\\[', '\\]')
                 break
             case 46:
-                replaceMathDelimitersInRange(document, range, '$', '\\(', '\\)')
+                this.replaceMathDelimitersInRange(document, range, '$', '\\(', '\\)')
                 break
             default:
                 break
         }
     }
+
+    private replaceWhitespaceOnLineBefore(document: vs.TextDocument, position: vs.Position, replaceWith: string) {
+        const beforePosRange = new vs.Range(new vs.Position(position.line, 0), position)
+        const text = document.getText(beforePosRange)
+        const regexResult = /\s*$/.exec(text)
+        if (!regexResult) {
+            return vs.workspace.applyEdit(new vs.WorkspaceEdit())
+        }
+        const charactersToRemove = regexResult[0].length
+        const wsRange = new vs.Range(new vs.Position(position.line, position.character - charactersToRemove), position)
+        const edit = new vs.WorkspaceEdit()
+        edit.replace(document.uri, wsRange, replaceWith)
+        return vs.workspace.applyEdit(edit)
+    }
+
+    private replaceRangeWithString(document: vs.TextDocument, range: vs.Range, replacementString: string) {
+        const edit = new vs.WorkspaceEdit()
+        edit.replace(document.uri, range, replacementString)
+        return vs.workspace.applyEdit(edit)
+    }
+
+    private replaceRangeWithRepeatedString(document: vs.TextDocument, range: vs.Range, replacementString: string) {
+        return this.replaceRangeWithString(document, range, replacementString.repeat(range.end.character - range.start.character))
+    }
+
+    private replaceMathDelimitersInRange(document: vs.TextDocument, range: vs.Range, oldDelim: string, startDelim: string, endDelim: string) {
```

Comment:
It would be better to change the type of `oldDelim` from `string` to `'$' | '$$'`.